docs/tasks/administer-cluster/configure-upgrade-etcd.md

```diff
@@ -16,47 +16,196 @@ redirect_from:
+
+Configure an etcd cluster either by static member information or by dynamic discovery. For more information on clustering, see [etcd Clustering Documentation](https://github.com/coreos/etcd/blob/master/Documentation/op-guide/clustering.md).
+
+For an example, consider a five-member etcd cluster running with the following client URLs: `http://$IP1:2379`, `http://$IP2:2379`, `http://$IP3:2379`, `http://$IP4:2379`, and `http://$IP5:2379`. To start a Kubernetes API server:
+
+1. Run the following:
+
+  ``` bash
+   ./etcd --client-listen-urls=http://$IP1:2379, http://$IP2:2379, http://$IP3:2379, http://$IP4:2379, http://$IP5:2379 --client-advertise-urls=http://$IP1:2379, http://$IP2:2379, http://$IP3:2379, http://$IP4:2379, http://$IP5:2379
+  ```
+
+2. Start Kubernetes API servers with the flag `--etcd-servers=$IP1:2379, $IP2:2379, $IP3:2379, $IP4:2379, $IP5:2379`.
+
+    Replace `IP` with your client IP addresses.
+
+###  Multi-node etcd cluster with load balancer
+
+To run a load balancing etcd cluster:
+
+1. Set up an etcd cluster.
+2. Configure a load balancer in front of the etcd cluster.
+   For example, let the address of the load balancer be `$LB`.
+3. Start Kubernetes API Servers with the flag `--etcd-servers=$LB:2379`.
+
+## Securing etcd clusters
+
+Access to etcd is equivalent to root permission in the cluster and ideally only the API server will have access to it. Considering the sensitivity of the data, it is recommended to grant permission to only those nodes that require access to etcd clusters.
+
+To secure etcd, either set up firewall rules or use the security features provided by etcd. etcd security features depend on x509 Public Key Infrastructure (PKI). First step is to establish secure communication channels by generating a key and certificate pair. For example, use key pairs `peer.key` and `peer.cert` for securing communication between etcd members, and `client.key` and `client.cert` for securing communication between etcd and its clients. See the [example scripts](https://github.com/coreos/etcd/tree/master/hack/tls-setup) provided by the etcd project to generate key pairs and CA files for client authentication.
+
+### Securing communication
+
+To configure etcd with secure peer communication, specify flags `--peer-key-file=peer.key` and `--peer-cert-file=peer.cert`, and use https as URL schema.
+
+Similarly, to configure etcd with secure client communication, specify flags `--key-file=peer.key` and `--cert-file=peer.cert`, and use https as URL schema.
+
+### Limiting access of etcd clusters
+
+After configuring secure communication, restrict the access of etcd cluster to only Kubernetes API server. Use TLS authentication to do so.
```

Comment:
suggested edit: "...restrict the access of the etcd cluster to only the Kubernetes API server."